### ISSUE TITLE
keep other sharer.yaml buttons parameters unchanged when changing one button parameter in a specific page

### ISSUE THREAD

**Issue by nicogreu** [ayozehd/sharer#4]
If in my sharer.yaml file, I have, say, facebook, twitter and pinterest enabled:
```
buttons:
    facebook:
        enabled: true
        icon: facebook
    twitter:
        enabled: true
        icon: twitter
    pinterest:
        enabled: true
        icon: pinterest

```
And then in the yaml of a specific page I want to provide the image to share on pinterest:

```
---
sharer:
    buttons:
        pinterest:
             image: http://pathToMyImage.jpg
---
```

It seems the "buttons:" reinitiates all the buttons option values of sharer.yaml and so nothing shows up unless I re-enter those options in the yaml of this specific page:

```
---
sharer:
    buttons:
        facebook:
            enabled: true
            icon: facebook
        twitter:
            enabled: true
            icon: twitter
        pinterest:
            enabled: true
            icon: pinterest
            image: http://pathToMyImage.jpg
---

```
Is there a way to enter specific button option in a page while keeping the default other options from the sharer.yaml file?

**Comment by ayozehd** [comment]
Thanks for your report @nicogreu, right now adding a custom image to the buttons in any of page-specific or plugin configuration it's not supported. The good news is I can manage to include this feature without major problems. I'll add it to TODO list asap :)

On the other hand the page-specific configuration NEVER should overwrite other values from plugin configuration, so this is **a important bug** to fix. For now you can customize: enabled, icon, label and priority.

**Comment by nicogreu** [comment]
Thanks @ayozehd ! Good to know that this overwrite behavior is a bug.

> right now adding a custom image to the buttons in any of page-specific or plugin configuration it's not supported

It seems to work actually. I mean I don't know if it is supposed to work, but I had a look at the javascript file, saw that there was this image option for pinterest and so I tried... and it seemed to work. Only thing was the overwriting bug

By the way, thanks for making this plugin! It's very useful for me :)

**Comment by ayozehd** [comment]
Ahh yes, I got what you mean now! The `image` value displays the picture in the message box, sharer.js automatically do the mapping on the platforms that supports it. I thought you mean about change the svg icon so only fixing the parameters override by page should be enough for you. All rest of values in buttons that are not managed by this plugin, are passed as `@data-values` to sharer.js

PS. Don't have too much time right now, but I will ping on this thread once I got a fix ;)
